lib/Console/Command/Parse.php

```diff
@@ -23,37 +23,93 @@
 
 use \Symfony\Component\Console\Input\InputInterface;
 use \Symfony\Component\Console\Output\OutputInterface;
+use \Mesamatrix\Parser\OglVersion;
+use \Mesamatrix\Parser\UrlCache;
+use \Mesamatrix\Parser\Hints;
 
 class Parse extends \Symfony\Component\Console\Command\Command
 {
-    protected function configure()
-    {
+    protected $urlCache;
+    protected $statuses;
+
+    protected function configure() {
         $this->setName('parse')
              ->setDescription('Parse data and generate XML')
         ;
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        //$gitLog = exec_git("log --pretty=format:%H --reverse " .
-        //    Mesamatrix::$config->getValue("git", "oldest_commit").".. -- ".Mesamatrix::$config->getValue("git", "gl3"), $log);
-        //$initialCommit = rtrim(fgets($log));
-        $initialCommit = "master";
-
-        $cat = new \Mesamatrix\Git\ProcessBuilder(array(
-          'show', $initialCommit.':'.\Mesamatrix::$config->getValue('git', 'gl3')
+    protected function execute(InputInterface $input, OutputInterface $output) {
+		$gl3Path = \Mesamatrix::$config->getValue('git', 'gl3', 'docs/GL3.txt');
+        $gitLog = new \Mesamatrix\Git\ProcessBuilder(array(
+            'log', '--pretty=format:%H|%at|%aN|%cN', '--reverse',
+            \Mesamatrix::$config->getValue('git', 'oldest_commit').'..', '--',
+            $gl3Path
         ));
-        $proc = $cat->getProcess();
-        $this->getHelper('process')->mustRun($output, $proc);
-
-        $hints = new \Mesamatrix\Parser\Hints();
-        $parser = new \Mesamatrix\Parser\OglParser($hints);
-        $matrix = $parser->parse_content($proc->getOutput());
+        $gitLogProc = $gitLog->getProcess();
+        $this->getHelper('process')->mustRun($output, $gitLogProc);
+
+        $commitLines = explode(PHP_EOL, $gitLogProc->getOutput());
+        $hints = new Hints();
+        $matrix = new \Mesamatrix\Parser\OglMatrix();
+        foreach ($commitLines as $commitLine) {
+            list($commitHash, $time, $author, $committer) = explode('|', $commitLine);
+            $commit = new \Mesamatrix\Git\Commit($commitHash, $time, $author, $committer);
+            \Mesamatrix::$logger->info('Parsing GL3.txt for commit '.$commit->getHash());
+            $cat = new \Mesamatrix\Git\ProcessBuilder(array(
+              'show', $commit->getHash().':'.$gl3Path
+            ));
+            $proc = $cat->getProcess();
+            $this->getHelper('process')->mustRun($output, $proc);
+
+            $parser = new \Mesamatrix\Parser\OglParser($hints, $matrix, $commit);
+            $parser->parse_content($proc->getOutput());
```

Comment:
I think it's weird to instantiate a new `OglParser` each time we want to parse the file. The parser is the same in any case, so it feels weird to see it being constructed and destroyed at each loop.

To prevent confusion, `parse_content()` should return void instead of the matrix given in the constructor. And in that note I think the matrix should not be given to the constructor since it's not really a part of the parser. To make things simpler to understand, I think `parse_content()` and `parse_stream()` should take the matrix as a parameter. In the end here are the declaration I would find simpler:

* parse_content($output, $matrix): void
* parse_stream($handle, $matrix): void

Plus the fact that the `OglParser` class won't be created at each loop, what would you think of that?